rclcpp_components/cmake/rclcpp_components_add_node.cmake

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Create a target which builds a shared library and also creates a target
+# which builds an executable
+# :param target: the executable name
+# :type target: string
+# :param ARGN: the list of source files for executable and shared library
+# :type ARGN: list of strings
+#
+function(rclcpp_components_add_node)
+
+  cmake_parse_arguments(
+    PARSED_ARGS
+    "EXCLUDE"
+    "TARGET"
+    "SRCS"
+    ${ARGN}
+    )
+
+  set(target ${PARSED_ARGS_TARGET})
+  set(srcs ${PARSED_ARGS_SRCS})
+  set(exclude ${PARSED_ARGS_EXCLUDE})
+
+  set(_sources_list)
+  foreach(_arg ${srcs})
```

Comment:
If the argument was passed as a multi-value keyword arg in `cmake_parse_arguments`, then this for-loop wouldn't be necessary ([see docs](https://cmake.org/cmake/help/v3.10/command/cmake_parse_arguments.html)). But, as per my previous comment, I think it's better to not parse them.